go/interpreter/lfvm/memory.go

```diff
@@ -171,29 +164,33 @@ func (m *Memory) SetWord(offset uint64, value *uint256.Int, c *context) error {
 	return nil
 }
 
-func (m *Memory) Set(offset, size uint64, value []byte) error {
+// set sets the given value at the given offset.
+// Returns error if insufficient memory or offset+size overflows.
+func (m *Memory) set(offset, size uint64, value []byte) error {
 	if size > 0 {
 		if offset+size < offset {
 			return errGasUintOverflow
 		}
 		if offset+size > m.length() {
-			return fmt.Errorf("memory too small, size %d, attempted to write %d bytes at %d", m.length(), size, offset)
+			return makeInsufficientMemoryError(m.length(), size, offset)
 		}
 		copy(m.store[offset:offset+size], value)
 	}
 	return nil
 }
 
-func (m *Memory) SetWithCapacityAndGasCheck(offset, size uint64, value []byte, c *context) error {
+func makeInsufficientMemoryError(memSize, size, offset uint64) error {
+	return tosca.ConstError(fmt.Sprintf("memory too small, size %d, attempted to write %d bytes at %d", memSize, size, offset))
+}
```

Comment:
To be honest, I am not a fan of introducing these error-factory functions with the main purpose of allowing tests to produce the same error for checking. It adds extra code and symbols in the namespace that should not be needed.

I think in this case a constant `errInsufficientMemory` would be enough, since no where in the code anybody would ever check for the memSize, size, or offset. It would require parsing the string, which is not a great idea. And human consumers will not greatly benefit from this information either.

The constant `errInsufficientMemory` constant could also be avoided if you would check for the error string to contain some "memory too small" in the test and ignore the rest. In this case, you could use `fmt.Errorf(..)` above to create the error string with the `memSize`, `size`, and `offset` as listed here. This eliminates the `errors.Is(...)`, but I think it is a cleaner solution if the error is not carrying any additional information for the caller than something went wrong and only in the test you want to distinguish what went wrong.

Reply:
discussed offline. I will go for the suggested `fmt.Errorf`.